File: pkg/server/pokete/msg/map_info/info.go
package map_info

import (
	"github.com/lxgr-linux/pokete/bs_rpc/msg"
	"github.com/lxgr-linux/pokete/resources"
	"github.com/lxgr-linux/pokete/server/context"
	"github.com/lxgr-linux/pokete/server/pokete/user"
)

const InfoType msg.Type = "pokete.map.info"

type Info struct {
	msg.BaseMsg
	Assets       resources.Assets `json:"assets"`
	Position     user.Position    `json:"position"`
	Users        []user.User      `json:"users"`
	GreetingText string           `json:"greeting_text"`
}

func (i Info) GetType() msg.Type {
	return InfoType
}

func NewInfo(
	assets resources.Assets,
	position user.Position,
	users context.Users,
	greetingtext string,
) Info {
	return Info{
		msg.BaseMsg{},
		assets,
		position,
		users.GetAllUsers(),
		greetingtext,
	}
}

File: src/pokete/classes/multiplayer/msg/map_info/info.py
from typing import TypedDict

import pokete.bs_rpc as bs_rpc
from pokete.classes.asset_service.resources import AssetsDict
from pokete.classes.multiplayer.msg.position import Position
from pokete.classes.multiplayer.msg.position.update import UpdateDict

INFO_TYPE = "pokete.map.info"


class InfoData(TypedDict):
    assets: AssetsDict
    position: Position
    users: list[UpdateDict]
    greeting_text: str


class Info(bs_rpc.Body):
    def __init__(self, data: InfoData):
        super().__init__(INFO_TYPE, data)
